src/services/data/schemas.ts

```diff
@@ -36,6 +36,28 @@ export default (client, httpAuth: HttpInstance): DataSchemasService => ({
     };
   },
 
+  async findAll(options?: { rql?: RQLString }): Promise<Schema[]> {
+    // Extra check is needed because this function is call recursively with updated RQL
+    // But on the first run, we need to set the limit to the max to optimize
+    const result: PagedResult<Schema> = await this.find({
+      rql:
+        options?.rql && options.rql.includes('limit(')
+          ? options.rql
+          : rqlBuilder(options?.rql).limit(50).build(),
```

Comment:
const DEFAULT_LIMIT = 50

Reply:
add console.log('WARNING: total amount is > 2000, be aware that this function can hog up resources')